### PR TITLE
mgr/dashboard: rbd mirroring promotion/demotion UI

### PR DESCRIPTION
**Pull request by pereman2**
This is a clone of https://github.com/ceph/ceph/pull/46524 which I closed by mistake :/

Fixes: https://tracker.ceph.com/issues/55860
Signed-off-by: Pedro Gonzalez Gomez [pegonzal@redhat.com](mailto:pegonzal@redhat.com)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by Sarthak0702 — APPROVED** [review]

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts [hunk 1]
```diff
@@ -195,14 +195,32 @@ export class RbdListComponent extends ListWithDetails implements OnInit {
         this.getInvalidNameDisable(selection) ||
         selection.first().schedule_info === undefined
     };
+    const promoteAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      click: () => this.actionPrimary(true),
+      name: this.actionLabels.PROMOTE,
+      visible: () =>
+        this.selection.first() != null && !this.selection.first().primary ? true : false
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts:204]: The conditional clause already evaluates to a boolean:
```suggestion
        this.selection.first() != null && !this.selection.first().primary
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts [hunk 2]
```diff
@@ -195,14 +195,32 @@ export class RbdListComponent extends ListWithDetails implements OnInit {
...
+    };
+    const demoteAction: CdTableAction = {
+      permission: 'update',
+      icon: Icons.edit,
+      click: () => this.actionPrimary(false),
+      name: this.actionLabels.DEMOTE,
+      visible: () =>
+        this.selection.first() != null && this.selection.first().primary ? true : false
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-list/rbd-list.component.ts:212]: Same here.

**Review by epuertat — APPROVED** [review]
LGTM!

**Comment by pereman2** [comment]
jenkins test api

**Comment by pereman2** [comment]
```

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
jenkins test dashboard cephadm

**Comment by pereman2** [comment]
jenkins test dashboard cephadm

**Comment by pereman2** [comment]
Failures look unrelated.
```
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  orchestrator/workflow/04-create-clu      00:04        2        1        1        -        - │
  │    ster-create-osds.e2e-spec.ts                                                                │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  orchestrator/workflow/05-create-clu      00:07        4        4        -        -        - │
  │    ster-review.e2e-spec.ts                                                                     │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  orchestrator/workflow/06-cluster-ch      01:22        9        9        -        -        - │
  │    eck.e2e-spec.ts                                                                             │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  orchestrator/workflow/07-osds.e2e-s      06:15        1        -        1        -        - │
  │    pec.ts                                                                                      │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  orchestrator/workflow/08-hosts.e2e-      01:09        4        4        -        -        - │
  │    spec.ts                                                                                     │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  orchestrator/workflow/09-services.e      03:30       10       10        -        -        - │
  │    2e-spec.ts                                                                                  │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✖  orchestrator/workflow/10-nfs-export      10:28        6        1        5        -        - │
  │    s.e2e-spec.ts                                                                               │
  └──────────────────────────
  ```

**Pull request merged**